src/client/components/pages/Courses/tableFields.tsx

```diff
@@ -143,6 +148,60 @@ export const formatMeetings = (
     id: courseId,
   } = course;
   const [modalVisible, setModalVisible] = useState(false);
+  /**
+   * Keeps track of whether the user has altered fields in the form to determine
+   * whether to show a confirmation dialog on modal close
+   */
+  const [
+    isChanged,
+    setIsChanged,
+  ] = useState(false);
+
+  const confirmMessage = "You have unsaved changes. Click 'OK' to disregard changes, or 'Cancel' to continue editing.";
+
+  useEffect(() => {
+    /**
+     * Checks to see if there are any unsaved changes in the modal When the user
+     * refreshes the page. If there are unsaved changes, the browser displays a
+     * warning message to confirm the page reload. If the user selects cancel, the
+     * user can continue making changes in the modal.
+     */
+    const onBeforeUnload = (event: Event) => {
+      if (!isChanged) return;
+      event.preventDefault();
+      // It's unclear whether TS will account for browser incompatibility here,
+      // so we use all three methods of setting the confirmation message,
+      // as detailed here:
+      // https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event
+      const legacyEvent = (event as unknown as LegacyEvent);
```

Comment:
I'm not totally clear on the need for the `LegacyEvent` -- is that for compatibility with a specific browser?

Reply:
Oh yes, so I saw this in the documentation for `beforeunload event`:
![image](https://user-images.githubusercontent.com/29589689/128200251-35730e87-4cb4-45ce-9fe5-ad876d0a2eff.png)

I was trying to account for the unsaved changes warning showing up in different browsers, so I initially tried to assign a return value property to the `event` by doing...
```
`event.returnValue = confirmMessage;`
```

...however, I got the eslint error: `Assignment to property of function parameter 'event'.` As a result, I created `legacyEvent`:
```
const legacyEvent = event;
legacyEvent.returnValue = confirmMessage;
```
The eslint error I got from the above when trying to assign `confirmMessage` to the `returnValue` property was `Type 'string' is not assignable to type 'boolean',` so this is what prompted me to create `LegacyEvent` as a workaround.